iDiff/differs/package.go

```diff
@@ -75,3 +64,95 @@ func getDiffOutput(d1file, d2file string, json bool) (string, error) {
 
 	return buffer.String(), nil
 }
+
+func diffImageFiles(img1, img2 string) (utils.DirDiff, error) {
+	var diff utils.DirDiff
+	img1FS, err := utils.ImageToFS(img1)
+	if err != nil {
+		return diff, fmt.Errorf("Error retrieving image %s file system: %s", img1, err)
+	}
+	img2FS, err := utils.ImageToFS(img2)
+	if err != nil {
+		return diff, fmt.Errorf("Error retrieving image %s file system: %s", img2, err)
+	}
+
+	img1Contents, err := getImageContents(img1FS)
+	if err != nil {
+		return diff, fmt.Errorf("Error parsing image %s contents: %s", img1, err)
+	}
+	img2Contents, err := getImageContents(img2FS)
+	if err != nil {
+		return diff, fmt.Errorf("Error parsing image %s contents: %s", img2, err)
+	}
+	defer os.RemoveAll(img1FS)
+	defer os.RemoveAll(img2FS)
+
+	for layer1, contents1 := range img1Contents {
+		sameLayer := false
+		for layer2, contents2 := range img2Contents {
+			if checkSameLayer(contents1, contents2) {
+				delete(img2Contents, layer2)
+				sameLayer = true
+				break
+			}
+		}
+		if sameLayer {
+			delete(img1Contents, layer1)
+		}
+	}
+
+	img1Dir := utils.Directory{
+		Root:    img1,
+		Content: getContentList(img1Contents),
+	}
+	img2Dir := utils.Directory{
+		Root:    img2,
+		Content: getContentList(img2Contents),
+	}
+	diff = utils.DiffDirectory(img1Dir, img2Dir)
+	return diff, nil
+}
+
+func getContentList(imgMap map[string]utils.Directory) []string {
+	contents := []string{}
+	for layer, dir := range imgMap {
+		for _, file := range dir.Content {
+			contents = append(contents, filepath.Join(layer, file))
+		}
+	}
+	return contents
+}
+
+func checkSameLayer(layer1, layer2 utils.Directory) bool {
```

Comment:
Yeah, or something similar. You might need to do some pre-processing to figure out which layers don't match up, but then you could just compare the ones that do side by side.